### ISSUE TITLE
Request - Different Layout of camera options

### ISSUE THREAD

**Issue by xyz4tech** [NoahFetz/F1AppleTV#28]
I really enjoy the app with the opportunity to add different camera options on one screen. However, I think it would be a nice thing to have one big camera option (e.g. main feed) and an amount of smaller cameras options (e.g. onboard cameras).

My request is to add a different layout to the app as additional option. The layout should have a bigger camera option with 75% of the screen and smaller camera options with a size of 25% of each. So it should be possible to add till four small camera options. In the settings of the app it could be possible to add a switch which allows to switch between the current layout with the equal sized screens and the the requested layout.

Example of my thoughts:
![F1TV](https://user-images.githubusercontent.com/68329729/117546331-36d55e00-b02a-11eb-8cc2-f3978ec173c9.png)

I really would love this layout.

Many thanks for your effort and time.

**Comment by NoahFetz** [comment]
This has been asked by a few times. I experimented with it in the beginning when I started making the multi view player. It's very difficult to get it right and will be something for the future when I have more time than now. I'm currently very busy with work stuff, that's why not many updates are coming.
If you want a detailed explaination why it's so difficult, you can go read the comment I made on issue #21

**Comment by xyz4tech** [comment]
As I understood the issue comes with the aspiration of a fully dynamic design. A premade layout based on pixels is critical because of different tv screens. For my screenshot I used a not fully dynamic approach but also not static. The following parameters are from the MPV player. It's maybe an approach for the app. It doesn't fit to the aspiration of a fully dynamic design but maybe it reduces the complexity for creating the layout.

Main Left
"--autofit-larger=74%", "-geometry=0:50%"

Sub Top Right
"--autofit-larger=26%", "-geometry=100%:14%"

Sub Middle Right
"--autofit-larger=26%", "-geometry=100%:50%"

Sub Bottom Right
"--autofit-larger=26%", "-geometry=100%:86%"

**Comment by BlackWolf** [comment]
I would love to see this, and as an iOS dev myself I might be able to spare some free time to implement this. I think, though, @NoahFetz is right - a fully dynamic design is a loot of work and can cause a lot of trouble. If this is acceptable, I think I could implement a static version of this - which means, as long as you have 2, 3 or 4 streams active, you can switch between the two layouts (grid and large first stream) in the channel menu.

Is this something you would be interested in @NoahFetz ?

**Comment by robbinonline** [comment]
Hi @NoahFetz is there any news about this, or do you need any help?

**Comment by BlackWolf** [comment]
> There's no point in doing a whole lot of work if their app ends up being better.

Fair point. Considering the state of their other apps, seems unlikely though :D

**Comment by robbinonline** [comment]
Is there any update on this. So for I experienced the f1 atv app is very basic so I don't see the functionality coming there.

**Comment by NoahFetz** [comment]
Nothing release ready yet

**Comment by Robsta86** [comment]
I would also love to see this featuer implemented, it would make this app even better than it already is :-)

**Comment by robbinonline** [comment]
is this app still being worked on?

**Comment by NoahFetz** [comment]
Yes but I'm pretty busy at the moment. I will try to add this for next season

**Comment by cryptedbv** [comment]
Any update? As the season nearly starts. ☺️

**Comment by NoahFetz** [comment]
I'm still working on it, probably won't make it for the first race :(

**Comment by cryptedbv** [comment]
Ah no problem, we appreciate all the effort! Enjoy this 2023 season! :D

**Comment by cryptedbv** [comment]
Hi Noah,

Any update on this? 😁

**Comment by cryptedbv** [comment]
A few races later, don't mean to push, but is this still in the pipeline? @NoahFetz

**Comment by SeaneyC** [comment]
I would be happy with a 3x3 or 4x4 grid design based on 16:9 aspect ratio windows. The “main” box takes up a 2x2 box in a 3x3 grid or a 3x3 box in a 4x4 grid and the other boxes can be used for any of the other screens or just left blank of course. 

For me the 3x3 grid with a 2x2 main window and 5 other streams in the remaining smaller windows would be perfect.

Not sure if there’s any options for PIP at all (or just one view laid over the top of the other)? If possible a full screen main image of 3x3 or 4x4 with a 2nd screen in a corner of your choice of size 1x1 would be amazing!

**Comment by cryptedbv** [comment]
So... anyone still going to do this? Bit bummer.
